invenio_pages/views.py

```diff
@@ -42,16 +41,6 @@
 )
 
 
-@blueprint.before_app_first_request
```

Comment:
Was this problem of registering the views solved?

Reply:
The problem of registering the views was solved once I tuned the deps a bit as I saw there was some webpack variables being injected in the context. This had no influence in that issue, but since we have the 404 handler that takes care of register the view, I just removed this one as it's deprecated.

Reply:
Ok, can we make it more explicit in the code in the `handle_not_found` func? We will forget about this.
In case of concurrency, what will happen when `_add_url_rule` will be called multiple times on the same URL path?

Reply:
Ok, will update the func in the next PR to be more explicit.
Regarding the `_add_url_rule`, I guess it should throw an error, I will test it and add error handling for this one, thanks a lot for pointing this out!